packages/common/src/utils/object/common.ts

```diff
@@ -92,3 +92,7 @@ export const filterEntries = pickBy;
 
 export const mapValue = <T, R>(obj: Record<string | number | symbol, T>, callback: (value: T) => R) =>
   Object.fromEntries(Object.entries(obj).map(([key, value]) => [key, callback(value)]));
+
+export const shallowPartialEquals = (obj: unknown, partial: unknown) => {
+  return isObject(obj) && isObject(partial) && Object.keys(partial).every((key) => partial[key] === obj[key]);
```

Comment:
```suggestion
  return isObject(obj) && isObject(partial) && Object.entries(partial).every(([key, partialValue]) => partialValue === obj[key]);
```

Reply:
I'm very surprised by this lol.
<img width="1028" alt="Screen Shot 2022-06-22 at 1 37 00 PM" src="https://user-images.githubusercontent.com/5643574/175131360-0d248f47-ace1-4621-9e37-5fe023ab176b.png">